DOCS-3499: Improve capture edge data (#3956)
viamrobotics/docs: docs/data-ai/capture-data/capture-sync.md

You can use the data management service to capture data from [supported components and services](#supported-resources), then sync it to the cloud.
You can also sync data from arbitrary folders on your machine.

Data capture and sync in Viam involves two key pieces:

- The data management {{< glossary_tooltip term_id="service" text="service" >}} that writes captured data to local edge device storage and syncs that data with the cloud.
- Individual {{< glossary_tooltip term_id="resource" text="resource" >}} configurations that specify what data to capture and how often.

## How data capture and data sync works

The data management service writes data from your configured Viam resources to local storage on your edge device and syncs data from the edge device to the cloud:

- The data management service stores captured data locally in <file>~/.viam/capture<file> by default.
- Data is synced to the Viam cloud at a configured sync interval using encrypted gRPC calls and deleted from the disk once synced.
- You can capture and sync data independently, one can run without the other.

> For more information, see [How sync works](data-ai/capture-data/advanced/how-sync-works/).

Author: ```suggestion
For more information, see [How sync works](/data-ai/capture-data/advanced/how-sync-works/).
```

You can use the data management service to capture data from [supported components and services](#supported-resources), then sync it to the cloud.
You can also sync data from arbitrary folders on your machine.

> ## Overview

Reviewer: We don't use headings like this anywhere in the docs. It's an explicit choice to not do this and instead place this type of content elsewhere - often at the top of the page, here it should go under the headline `Set up data capture and data sync` where there's no content before the next heading.

```suggestion
```

Author: ok!

Data capture and sync in Viam involves two key pieces:

> - A data management {{< glossary_tooltip term_id="service" text="service" >}} that writes captured data to local edge device storage and syncs that data with the cloud.

Reviewer: Unless this has changed, there can only be one data management service, that's why it should be a definite article.

Author: good point

- Individual {{< glossary_tooltip term_id="resource" text="resource" >}} configurations that specify what data to capture and how often.

> ## How data capture and data sync works

Reviewer: Please make sure there is at least an intro sentence when you have bullet points. 

Worth considering if the explanation of how it works is actually what's needed this early in the page. Do we expect people to come here primarily for setup or for explanation?

Author: IMO the old page was very setup oriented so I think more of an explanation is worth trying and presenting to eliot

Reviewer 2: I don't view them as mutually exclusive; imo a quick overview / explanation of the system before getting into specific set up steps helps contextualize the set up work

Reviewer: Sounds good. It's always a trade off between context and pushing the content people may be trying to get to below the fold. If you both in the future want to check data on this you can always look at clarity data.

only other consideration is whether you want to link to the more detailed how capture & sync works page. Please feel free to resolve once you've considered this

- The data management service stores captured data locally in <file>~/.viam/capture<file> by default.
- Data is synced to the Viam cloud at a configured sync interval using encrypted gRPC calls and deleted from the disk once synced.
- You can capture and sync data independently, one can run without the other.

> ## Set up data capture and data sync

Reviewer: Headings should never be followed by another heading like this without text in between. This generally indicates a content design issue. 

In this case I wouldn't be surprised it you'd also find that people accidentally skip the step to add the data mgmt service. There is nothing helping the reader to avoid this from happening so when they're just skimming, they could easily miss it.

Author: thank you for the explanation!

### Configure a data management service for your machine

The data management service is responsible for:

- Writing data from your configured Viam resources to local storage on your edge device

> - Syncing data from the edge device to the cloud

Reviewer: This is explanatory content that you've put with a heading where readers would expect pragmatic content. This info should likely be with the "How data capture and data sync works" section.

| `cache_size_kb` | float | Optional | `viam-micro-server` only. The maximum amount of storage bytes (in kilobytes) allocated to a data collector. <br> Default: `1` KB. |  <p class="center-text"><i class="fas fa-check" title="yes"></i></p> |
| `file_last_modified_millis` | float | Optional | The amount of time to pass since arbitrary files were last modified until they are synced. Normal <file>.capture</file> files are synced as soon as they are able to be synced. <br> Default: `10000` milliseconds. |  <p class="center-text"><i class="fas fa-times" title="no"></i></p> |

{{< /expand >}}

### Configure data capture for individual resources

You can capture data for any {{< glossary_tooltip term_id="resource" text="resource" >}} that supports it, including resources on {{< glossary_tooltip term_id="remote-part" text="remote parts" >}}.

> Configure data capture for individual resources by:

Reviewer: I think you need to add where they'd do this though.

Author: I think those instructions are right below in the tabs. Do we need to repeat them twice?

Author: still adding

Reviewer: thx

{{< /expand >}}

### Supported resources

The following components and services support data capture and cloud sync:

{{< readfile "/static/include/data/capture-supported.md" >}}

## Stop data capture

If you don't need to capture data, for instance in a test scenario, you can turn off data capture to reduce unnecessary storage.

> Alternatively, see [advanced data capture and sync configurations](/data-ai/capture-data/capture-sync/#advanced-data-capture-and-sync-configurations) for other ways to control data usage, such as conditional sync or retention policies.

Reviewer: nice

Author: @jdamon96 did you want to keep this section above? I'm in favor of keeping it along with the text you added below

Reviewer 2: yes - along w/ the text added - agree!

{{< /expand >}}

### Supported resources

The following components and services support data capture and cloud sync:

{{< readfile "/static/include/data/capture-supported.md" >}}

## View captured data

To view all the captured data you have access to, go to the [**DATA** tab](https://app.viam.com/data/view) where you can filter by location, type of data, and more.

You can also access data from a resource or machine part menu.

## Stop data capture

If you don't need to capture data, for instance in a test scenario, you can turn off data capture to reduce unnecessary storage.
Alternatively, see [advanced data capture and sync configurations](/data-ai/capture-data/capture-sync/#advanced-data-capture-and-sync-configurations) for other ways to control data usage, such as conditional sync or retention policies.

> To turn off data capture for a specific resource's capture method (for example, a camera component capturing via the `GetImage` capture method) navigate to the **Data capture** section of your resource's configuration card and toggle the configured capture method's switch to **Off**.

Author: ```suggestion
To turn off data capture for a specific resource's capture method (for example, a camera component capturing through the `GetImage` capture method) navigate to the **Data capture** section of your resource's configuration card and toggle the configured capture method's switch to **Off**.
```

You can also access data from a resource or machine part menu.

## Advanced data capture and sync configurations

> ### Direct to MongoDB capture

Reviewer: I don't understand this headline - maybe "Capture directly to MongoDB"?

You can configure direct capture of tabular data to a MongoDB instance alongside disk storage on your edge device.
This can be useful for powering real-time dashboards before data is synced from the edge to the cloud.

> Configure using `mongo_capture_config` attributes in your data manager service.

Reviewer: grammar?

> Configure using `mongo_capture_config` attributes in your data manager service.

Author: ```suggestion
Configure using the `mongo_capture_config` attributes in your data manager service.
```

Reviewer 2: tiny nit - now that we've added the 'the', it should be _attribute_ (singular) rather than attribute**s**

If writes to MongoDB fail for any reason, data capture will log an error for each failed write and continue capturing.

Failing to write to MongoDB doesn't affect capturing and syncing data to cloud storage other than adding capture latency.

{{< alert title="Caution" color="caution" >}}

- Capturing directly to MongoDB may write data to MongoDB that later fails to be written to disk (and therefore never gets synced to cloud storage).
- Capturing directly to MongoDB does not retry failed writes to MongoDB. As a consequence, it is NOT guaranteed all data captured will be written to MongoDB.
  This can happen in cases such as MongoDB being inaccessible to `viam-server` or writes timing out.
- Capturing directly to MongoDB may reduce the maximum frequency that data capture can capture data due to the added latency of writing to MongoDB.
  If your use case needs to support very high capture rates, this feature may not be appropriate.

{{< /alert >}}

### Conditional sync

> Although data sync checks for new data at the configured interval, you can define additional conditions that must be met to trigger syncing.

Reviewer: `Although` doesn't make sense here. There's no contradiction in that - rather it works on top of this: 

```suggestion
By default, `viam-server` checks for new data to sync at the configured interval (`sync_interval_mins `).
You can additionally configure sync to only happen when certain conditions are met.
```

For example:

- Only sync when on WiFi
- Sync when specific events are detected
- Sync during certain time windows

See [Conditional cloud sync](/data-ai/capture-data/conditional-sync/) for how to implement conditional syncs.

### Cloud data retention

Configure how long your synced data remains stored in the cloud:

> - **Retain data up to a certain size or for a specific length of time:** Set `retention_policies` at the resource level.
>   - See the `retention_policy` field in [data capture configuration attributes](/data-ai/capture-data/capture-sync/#click-to-view-data-capture-attributes).
> - **Delete data captured by a machine when you delete the machine:** Control whether your cloud data is deleted when a machine or machine part is removed.
>   - See the `delete_data_on_part_deletion` field in the [data management service configuration attributes](/data-ai/capture-data/capture-sync/#click-to-view-data-management-attributes).

Reviewer: This will look a bit better if you remove the additional level or nesting. However you could also consider using the exact same pattern as the one in the Sync optimization content and use no bullets.
```suggestion
- **Retain data up to a certain size or for a specific length of time:** Set `retention_policies` at the resource level.
  See the `retention_policy` field in [data capture configuration attributes](/data-ai/capture-data/capture-sync/#click-to-view-data-capture-attributes).
- **Delete data captured by a machine when you delete the machine:** Control whether your cloud data is deleted when a machine or machine part is removed.
    See the `delete_data_on_part_deletion` field in the [data management service configuration attributes](/data-ai/capture-data/capture-sync/#click-to-view-data-management-attributes).
```
